test/cluster/upsert/01-create-sources.td

```diff
@@ -50,9 +50,16 @@ $ set schema={
     ]
   }
 
-$ kafka-create-topic topic=dbzupsert partitions=1
+$ kafka-create-topic topic=dbzupsert-broken-key partitions=1
 
-$ kafka-ingest format=avro topic=dbzupsert key-format=avro key-schema=${keyschema} schema=${schema}
+$ kafka-ingest format=avro topic=dbzupsert-broken-key key-format=avro key-schema=${keyschema} schema=${schema}
```

Comment:
It's unclear to me if these changes are also related to the row ordering or it's just a cleanup. If they are affected by the row ordering can you explain how?

Reply:
This is explained in the other above-linked [Slack msg](https://materializeinc.slack.com/archives/C01LKF361MZ/p1725910128130889): The problem is that the test is relying on a certain error being shown when there are two errors in the collection's error stream. There is no guarantee about which error is shown when there are multiple errors in an error stream.

So, I've split the dbzupsert topic into two topics, so that the two errors can be tested separately. All changes are a consequence of this change.

Reply:
Makes sense, thanks